src/useSpotSell.ts

```diff
@@ -0,0 +1,153 @@
+    provider,
+    synthMarketId,
+    settlementStrategyId,
+  });
+  const { data: priceData } = useSpotGetPriceData({ provider, synthMarketId });
+
+  return useMutation({
+    mutationFn: async (amount: ethers.BigNumber) => {
+      if (
+        !(
+          chainId &&
+          SpotMarketProxyContract?.address &&
+          MulticallContract?.address &&
+          PythERC7412WrapperContract?.address &&
+          walletAddress &&
+          synthMarketId &&
+          synthTokenAddress &&
+          systemToken &&
+          provider &&
+          priceIds &&
+          spotSettlementStrategy &&
+          priceData
+        )
+      ) {
+        throw 'OMFG';
+      }
+
+      if (amount.lte(0)) {
+        throw new Error('Amount required');
+      }
+
+      const freshBalance = await fetchTokenBalance({
+        provider,
+        ownerAddress: walletAddress,
+        tokenAddress: synthTokenAddress,
+      });
+
+      if (freshBalance.lt(amount)) {
+        throw new Error('Not enough balance');
+      }
+
+      const freshAllowance = await fetchTokenAllowance({
+        provider,
+        ownerAddress: walletAddress,
+        tokenAddress: synthTokenAddress,
+        spenderAddress: SpotMarketProxyContract.address,
+      });
+
+      if (freshAllowance.lt(amount)) {
+        await fetchApproveToken({
+          provider,
+          walletAddress,
+          tokenAddress: synthTokenAddress,
+          spenderAddress: SpotMarketProxyContract.address,
+          allowance: amount.sub(freshAllowance),
+        });
+      }
+
+      const freshPriceUpdateTxn = await fetchPriceUpdateTxn({
+        provider,
+        MulticallContract,
+        PythERC7412WrapperContract,
+        priceIds: [spotSettlementStrategy.feedId],
+        stalenessTolerance: priceData.strictPriceStalenessTolerance,
+      });
+      console.log('freshPriceUpdateTxn', freshPriceUpdateTxn);
+
+      if (freshPriceUpdateTxn.value) {
+        console.log('-> fetchSpotSellWithPriceUpdate');
+        await fetchSpotSellWithPriceUpdate({
+          provider,
+          walletAddress,
+          SpotMarketProxyContract,
+          MulticallContract,
+          marketId: synthMarketId,
+          amount,
+          priceUpdateTxn: freshPriceUpdateTxn,
+        });
+        return { priceUpdated: true };
+      }
+
+      console.log('-> fetchSpotSell');
+      await fetchSpotSell({
+        provider,
+        walletAddress,
+        SpotMarketProxyContract,
+        marketId: synthMarketId,
+        amount,
+      });
+      return { priceUpdated: false };
+    },
+    throwOnError: (error) => {
+      // TODO: show toast
+      errorParser(error);
+      return false;
+    },
+    onSuccess: ({ priceUpdated }) => {
+      if (!queryClient) return;
+
+      if (priceUpdated) {
+        queryClient.invalidateQueries({
+          queryKey: [chainId, 'PriceUpdateTxn', { priceIds: priceIds?.map((p) => p.slice(0, 8)) }],
```

Comment:
We only update one price in this mutation (taken from settlement strategy), so we cannot clean cache for all prices

In this hook we font need priceIds and priceUpdateTxn

Reply:
Invalidate queries without slicing price IDs